docs/miscellaneous.rst

```diff
@@ -303,6 +303,124 @@ The following is the order of precedence for operators, listed in order of evalu
+
+Signed integers are represented in two's complement, and are restricted to the range ``[-2**(N - 1), 2**(N - 1) - 1]``, , where ``N`` is the bitsize.
+
+Overflow and underflow in signed and unsigned integers are generally handled by reducing the value ``modulo N``, where ``N`` is the bitsize of the integer in question, although there are some important exceptions (see the **special cases** section below).
+
+.. warning::
+    Solidity does not automatically detect over/underflow.
+
+Additionally, the EVM has its own overflow semantics for results that exceed 256 bits. See: `The Ethereum Yellow Paper <https://ethereum.github.io/yellowpaper>`_, Appendix H, Section 2 (Instruction set).
+
+Examples:
+
+``uint8(257) == 257 mod 2**8 == 1``
+
+``uint8(2 - 5) == -3 mod 2**8 == 256 - 3 == 253``
+
+``int8(-129) == -129 mod 256 == 127``
+
+**Casting and demotion**
+
+When casting a value into an integer type, the result is always reduced `modulo N`, where `N` is the bit-size of that integer type. This is important to remember when demoting an integer to a new type with a smaller bit-size.
+
+Example: ``uint8(257) = 1``;
+
+**Special cases**
+
+The range of signed integer types, i.e. ``[-2**(N - 1), 2**(N - 1) - 1]``, is asymmetric around 0. As a result of this, inverting the minimum value will result in a value that is ``1`` greater then the maximum value, making it wrap around back to itself:
+
+``-INT_MIN = -(-2**(N - 1)) = 2**(N - 1) = 2**(N - 1) - 1 + 1 = INT_MAX + 1 = INT_MIN``
+
+Table I: Rules for operators and unsigned Integers
+--------------------------------------------------
+
++------------+----------+-----------+---------+
+|  Operator  | Overflow | Underflow | Notes   |
++============+==========+===========+=========+
+| +, ++, +=  | x        |           |         |
++------------+----------+-----------+---------+
+| \+ (unary) | N/A      | N/A       | 1       |
++------------+----------+-----------+---------+
+| -, --, -=  |          | x         |         |
++------------+----------+-----------+---------+
+| \- (unary) |          |           | 2       |
++------------+----------+-----------+---------+
+| \*, \*=    | x        |           |         |
++------------+----------+-----------+---------+
+| \*\*       | x        |           | 3       |
++------------+----------+-----------+---------+
+| /, /=      |          |           |         |
++------------+----------+-----------+---------+
+| %, %=      |          |           |         |
++------------+----------+-----------+---------+
+| <<, <<=    |          |           | 4       |
++------------+----------+-----------+---------+
+| >>, >>=    |          |           | 5       |
++------------+----------+-----------+---------+
+| &, &=      |          |           |         |
++------------+----------+-----------+---------+
+| \|, \|=    |          |           |         |
++------------+----------+-----------+---------+
+| ^, ^=      |          |           |         |
```

Comment:
Strictly, this can also overflow or underflow. We just don't call it like that because it is a bit operator.

Perhaps it would be better to talk about the operators in paragraphs instead of a table?

Reply:
could use N/A for bit operators, to emphasize the difference. Btw I included arithmetic shift as an arithmetic operation since it is a form of hybrid.

I don't understand what paragraphs means. Does it mean removing the table and instead give every group of operators its own paragraph?

Reply:
this is another big issue, like introducing invariants. i will just leave this as-is for the next push, which i will do in a little while. I sense this will go down the arithmetic/bitwise rabbit hole 😧.